### PR TITLE
feat(push): add support for AuthKey

### PR DESCRIPTION
**Pull request by eoger**
See https://github.com/mozilla/fxa-auth-server/pull/1229 for context

### CODE REVIEW

File: client/FxAccountClient.js [hunk 1]
```diff
@@ -1012,9 +1023,11 @@ define([
           type: deviceType
         };
 
-        if (options.deviceCallback && options.devicePublicKey) {
+        if (options.deviceCallback && options.devicePublicKey
+            && options.deviceAuthKey) {
```
Comment (vladikoff) [inline client/FxAccountClient.js]: I'm wondering if we need to refactor this. I think there are cases where we update the deviceCallback without the publickey and the deviceauthkey. Seems like the signup above is doing it correctly.

**Comment by eoger** [comment]
Updated to follow the spec defined here : https://github.com/mozilla/fxa-auth-server/pull/1229#discussion-diff-60333579

**Comment by vladikoff** [comment]
Awesome thanks, let me know if this is still WIP or should be good to go.

**Comment by eoger** [comment]
GTG 👍

**Pull request merged**